### PR TITLE
Translation of some core templates(2020) to Hindi

### PR DESCRIPTION
**Pull request by PraveenPal4232**
Translation of some of these core templates(2020) to Hindi
- /2019/base.html
- /2019/contributors.html
- /2019/index.html
- /2019/table_of_contents.html
- /2019/errors.html
- /2020/base.html
- /2020/contributors.html
- /2020/index.html
- /2020/table_of_contents.html

Also enabled Hindi and added myself to config.

`methodology.html` is left in this folder. This is a large file, so I decided to raise a separate PR for that.

For issue #1402

### CODE REVIEW

File: src/templates/hi/2019/base.html [hunk 1]
```diff
@@ -0,0 +1,192 @@
+{% extends "base/2019/base.html" %}
```
Comment (bazzadp) [inline src/templates/hi/2019/base.html:1]: Has this changed? We've already merged this already?

Can you do a merge of main into your branch to bring them back in sync?

File: src/templates/hi/2020/base.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{% extends "%s/2019/base.html" % lang %}
+
+{% block methodology_stat_1 %}7.5M{% endblock %}
+{% block methodology_stat_2 %}31.3 TB{% endblock %}
+{% block methodology_description %}
+जब तक अन्यथा नोट न किया जाए, Web Almanac के सभी 22 अध्यायों में मौजूद मेट्रिक्स HTTP Archive के स्रोतों से लिये जाते हैं। HTTP Archive एक समुदाय द्वारा संचालित परियोजना है, जो 2010 के बाद से वेब कैसे बनाया जा रहा है इस पर निगाह रखें हुए हैं। WebPageTest और Lighthouse का अति उत्कृष्ट उपयोग करते हुए, 7.5 मिलियन से अधिक वेबसाइटों के बारे में मेटाडेटा को मासिक रूप से परीक्षण किया जाता है और विश्लेषण के लिए एक सार्वजनिक BigQuery डेटाबेस में शामिल किया जाता है। अगस्त 2020 डेटासेट का उपयोग Web Almanac के मैट्रिक्स के आधार के रूप में किया गया था। अधिक जानकारी के लिए, कार्यप्रणाली पृष्ठ देखें।
+{% endblock %}
+
+{% block foreword %}
+{% endblock %}
```
Comment (bazzadp) [inline src/templates/hi/2020/base.html:10]: Nit but can we have a newline at the end of files please.

```suggestion
{% endblock %}

```

File: src/templates/hi/2020/contributors.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+{% block description %}{{ config.contributors.items() | length }} लोग जिन्होंने {{ year }} Web Almanac में विश्लेषक, लेखक, डिजाइनर, डेवलपर्स, संपादक, लीडर, समीक्षक और अनुवादक के रूप में योगदान दिया।{% endblock %}
+
+{% block date_published %}2020-07-14T00:00:00.000Z{% endblock %}
+{% block date_modified %}2020-11-03T00:00:00.000Z{% endblock %}
+
+{% block filter_by_team %}टीम द्वारा फ़िल्टर करें: <span id="filtered-contributors">{{ self.contributors() }}</span><span id="contributors-total-text" class="hidden"> में से <span id="contributors-total">{{ config.contributors.items() | length }}</span></span> योगदानकर्ता.{% endblock %}
+{% block filter_by %}फ़िल्टर करें{% endblock %}
+
+{% block join_the_team_title%}Web Almanac टीम में शामिल हों{% endblock %}
+{% block join_the_team_text%}टीम में शामिल हों!{% endblock %}
```
Comment (bazzadp) [inline src/templates/hi/2020/contributors.html:14]:
```suggestion
{% block join_the_team_text%}टीम में शामिल हों!{% endblock %}

```

File: src/templates/hi/2020/index.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+{% block date_modified %}2020-11-03T00:00:00.000Z{% endblock %}
+
+{% block intro_title %}{{ self.coming_soon() }}{% endblock %}
+{% block intro_sub_title %}{{ year }} <b>वेब की स्थिति</b> की रिपोर्ट{% endblock %}
+
+{% block mission %}
+<p>
+  वेब समुदाय के विशेषज्ञ वर्तमान में Web Almanac के {{ year }} संस्करण के लिए कार्य योजना, विश्लेषण, और लेखन सामग्री पर कार्य कर रहे हैं, जो वर्ष के अंत में प्रकाशित होने के लिए ट्रैक पर है।
+</p>
+{% endblock %}
```
Comment (bazzadp) [inline src/templates/hi/2020/index.html:18]:
```suggestion
{% endblock %}

```

File: src/templates/hi/2020/table_of_contents.html [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{% extends "base/2019/table_of_contents.html" %}
+
+{% block title %}विषय - सूची | Web Almanac 2020{% endblock %}
+
+{% block description %}2020 Web Almanac के लिए सामग्री की तालिका के, प्रत्येक अनुभाग सूचीबद्ध है: पृष्ठ सामग्री, उपयोगकर्ता अनुभव, सामग्री प्रकाशन, सामग्री वितरण।{% endblock %}
+
+{% block twitter_image_alt %}{{ year }} Web Almanac की कार्यप्रणाली{% endblock %}
+
+{% block date_published %}2020-10-24T00:00:00.000Z{% endblock %}
+{% block date_modified %}2020-11-03T00:00:00.000Z{% endblock %}
```
Comment (bazzadp) [inline src/templates/hi/2020/table_of_contents.html:10]:
```suggestion
{% block date_modified %}2020-11-03T00:00:00.000Z{% endblock %}

```

File: src/templates/ja/2020/table_of_contents.html [hunk 6]
```diff
@@ -7,4 +7,4 @@
 {% block twitter_image_alt %}{{ year }} Web Almanac方法論{% endblock %}
 
 {% block date_published %}2020-10-24T00:00:00.000Z{% endblock %}
-{% block date_modified %}2020-11-03T00:00:00.000Z{% endblock %}
+{% block date_modified %}2020-11-03T00:00:00.000Z{% endblock %}
```
Comment (bazzadp) [inline src/templates/ja/2020/table_of_contents.html:10]:
```suggestion
{% block date_modified %}2020-11-03T00:00:00.000Z{% endblock %}

```

**Review by bazzadp — CHANGES_REQUESTED** [review]
Some nitpicks about line endings but main review is to merge last `main` from upstream into this.

Also it's now at the point that it's almost ready to send live, and so tests are getting confused why it's not live, so need to finish off some of the core templates before we accept this. So can we add the following translations to this PR:

- /2019/base.html
- /2019/contributors.html
- /2019/index.html
- /2019/table_of_contents.html

They should be very similar to 2020 versions anyway.

We also need the following to allow this to run:

- /2019/error.html
- change to `src/config/2019.json` to add `hi` to the `supported_languages` array
- change to `src/config/2020.json` to add `hi` to the `supported_languages` array
- Addition of Hindi to `src/server/language.py` file around line 34.

And also can you add an entry to both json files with your details, so you can be listed on the site? It's in alphabetical order by first name so around line 878 for 2019.json and line 1129 for 2020.json. You should be in the "translators" team for both.

File: src/templates/hi/2019/base.html (hunk 1 above)
Comment (PraveenPal4232) [reply src/templates/hi/2019/base.html]: Thanks, did that :)

File: src/templates/hi/2020/base.html (hunk 2 above)
Comment (PraveenPal4232) [reply src/templates/hi/2020/base.html]: Sure

File: src/templates/hi/2020/contributors.html (hunk 3 above)
Comment (PraveenPal4232) [reply src/templates/hi/2020/contributors.html]: Done!

File: src/templates/hi/2020/index.html (hunk 4 above)
Comment (PraveenPal4232) [reply src/templates/hi/2020/index.html]: Done!

File: src/templates/hi/2020/table_of_contents.html (hunk 5 above)
Comment (PraveenPal4232) [reply src/templates/hi/2020/table_of_contents.html]: Done!

File: src/templates/ja/2020/table_of_contents.html (hunk 6 above)
Comment (PraveenPal4232) [reply src/templates/ja/2020/table_of_contents.html:10]: Done!

Comment (bazzadp) [reply src/templates/ja/2020/table_of_contents.html:10]: Don't see this one fixed?

**Comment by bazzadp** [comment]
> Addition of Hindi to `src/server/language.py` file around line 34.

Don't see this one. Can you add that?

**Comment by bazzadp** [comment]
Oh and one more thing. The error.html file have translated error messages. See the French one for example: https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/src/templates/fr/2019/error.html

**Comment by PraveenPal4232** [comment]
Hi @bazzadp, How now everything is fine :)
Please let me know...

File: src/templates/hi/2019/error.html [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+{%
+  set localizedErrorMessage = {
+    "Unsupported year requested": "असमर्थित वर्ष का अनुरोध किया गया",
+    "Unsupported chapter requested": "असमर्थित अध्याय का अनुरोध किया",
+    "Unsupported language requested": "असमर्थित भाषा का अनुरोध किया",
+    "Not Found": "अनुरोधित URL सर्वर पर नहीं मिला। यदि आपने स्वयं URL दर्ज किया है तो कृपया अपनी वर्तनी की जाँच करें और पुनः प्रयास करें।"
+  }
+%}
+{% macro expandedErrorMessage(errorCode, errorDescription, errorMessage) %}
+{{ errorCode }} {{ errorDescription }}: {{ errorMessage }}
```
Comment (bazzadp) [inline src/templates/hi/2019/error.html:26]: One small thing here. As we show the English error message as well as the local one, let's add a `<i lang="en">` around that:

```suggestion
{{ errorCode }} <i lang="en">{{ errorDescription }}</i>: {{ errorMessage }}
```

**Review by bazzadp — APPROVED** [review]
That's looking great now @PraveenPal4232 . Just one tiny extra change.

I've put a test version here: https://20201110t142408-dot-webalmanac.uk.r.appspot.com/hi/2020/

You can also see [what your profile will look like on this](https://20201110t142408-dot-webalmanac.uk.r.appspot.com/hi/2020/contributors#PraveenPal4232)!

Could you have a surf around that and make sure you're happy with this? If so let me know and I'll merge this and it will be included in next release in next few days!

File: src/templates/hi/2019/error.html (hunk 7 above)
Comment (PraveenPal4232) [reply src/templates/hi/2019/error.html]: Done :)

**Comment by PraveenPal4232** [comment]
> That's looking great now @PraveenPal4232 . Just one tiny extra change.
> 
> I've put a test version here: https://20201110t142408-dot-webalmanac.uk.r.appspot.com/hi/2020/
> 
> You can also see [what your profile will look like on this](https://20201110t142408-dot-webalmanac.uk.r.appspot.com/hi/2020/contributors#PraveenPal4232)!
> 
> Could you have a surf around that and make sure you're happy with this? If so let me know and I'll merge this and it will be included in next release in next few days!

Just noticed a small issue in the contributors filter 
Now : टीम द्वारा फ़िल्टर करें: 15 में से 98 योगदानकर्ता।
Should be : टीम द्वारा फ़िल्टर करें: 98 में से 15 योगदानकर्ता।

Let me fix this before merge :) Rest is fine

**Comment by bazzadp** [comment]
OK. Let me know when good to go.

On a related note, I'm about to add a Chapter Links title:

Chapter Links Screenshot

Could you let me know what I should use for "Chapter Links" in Hindi? Google Translate suggests अध्याय लिंक - is that accurate?

**Comment by PraveenPal4232** [comment]
> Chapter Links



> OK. Let me know when good to go.
> 
> On a related note, I'm about to add a Chapter Links title:
> 
> Chapter Links Screenshot
> 
> Could you let me know what I should use for "Chapter Links" in Hindi? Google Translate suggests अध्याय लिंक - is that accurate?

Yes It is "अध्याय लिंक"

**Comment by bazzadp** [comment]
Here's what that looks like now:

Hindi contributors

**Comment by PraveenPal4232** [comment]
> Here's what that looks like now:
> 
> Hindi contributors

Yes! I checked it's perfect :)

**Comment by PraveenPal4232** [comment]
Thanks, @bazzadp for merging this PR

**Comment by bazzadp** [comment]
Thank you for submitting it!

**Pull request merged**